exporter/src/exporter.rs

```diff
@@ -0,0 +1,77 @@
+use trustification_event_bus::{Event, EventBus, EventConsumer};
+use trustification_storage::{EventType, Storage};
+
+pub async fn run<E: EventBus, M: Emitter + Send + Sync>(
+    storage: Storage,
+    bus: E,
+    emitter: M,
+    stored_topic: &str,
+    exported_topic: &str,
+    failed_topic: &str,
+) -> Result<(), anyhow::Error> {
+    let consumer = bus.subscribe("exporter", &[stored_topic]).await?;
+    loop {
+        select! {
+            event = consumer.next() => match event {
+                Ok(Some(event)) => {
+                    if let Some(payload) = event.payload() {
+                        if let Ok(data) = storage.decode_event(&payload) {
+                            for data in data.records {
+                                if data.event_type() == EventType::Put {
+                                    if storage.is_index(data.key()) {
+                                        tracing::trace!("It's an index event, ignoring");
+                                    } else {
+                                        let key = data.key();
+                                        match storage.get_for_event(&data).await {
+                                            Ok(data) => {
+                                                if let Ok(_) = bombastic_index::SBOM::parse(&data) {
+                                                    let document = Document {
+                                                        blob: data,
+                                                        r#type: DocumentType::UNKNOWN,
+                                                        format: FormatType::UNKNOWN,
+                                                        source_information: SourceInformation {
+                                                            collector: "S3Collector".into(),
+                                                            source: key.to_string(),
+                                                        },
+                                                    };
+
+                                                    match emitter.publish(document).await {
+                                                        Ok(_) => {
+                                                            tracing::trace!("Inserted entry into index");
+                                                            bus.send(exported_topic, key.as_bytes()).await?;
+                                                        }
+                                                        Err(e) => {
+                                                            let failure = serde_json::json!( {
+                                                                "key": key,
+                                                                "error": e.to_string(),
+                                                            }).to_string();
+                                                            bus.send(failed_topic, failure.as_bytes()).await?;
```

Comment:
Failed is primarily reserved for index failures, I'm not sure an export failure should be considered a failure, but perhaps make this optional if a topic is provided for notifying failures.